src/Http/Http.Extensions/src/RequestDelegateFactory.cs

```diff
@@ -819,5 +826,29 @@ private static ILogger GetLogger(HttpContext httpContext)
                 return loggerFactory.CreateLogger(typeof(RequestDelegateFactory));
             }
         }
+
+        private static void EnsureRequestTaskResultNotNull<T>(Task<T> task)
+        {
+            if (task == null)
+            {
+                throw new InvalidOperationException("Task Result should not be null");
+            }
+        }
+
+        private static void EnsureRequestValueTaskResultNotNull<T>(ValueTask<T> task)
```

Comment:
This one doesn't make sense since ValueTask can't be null (it's a struct). You can remove this check.

Reply:
For the ValueTask<IResult>, should we check that IResult is not null ?

Reply:
That's correct. Also IResult directly.